tests/unit/lms/resources/lti_launch_test.py

```diff
@@ -158,6 +101,60 @@ def test_it_returns_the_js_config(self, pyramid_request, JSConfig):
         assert js_config == JSConfig.return_value
 
 
+class TestShouldUseSectionGroups:
+    @pytest.mark.parametrize(
+        "is_canvas,feature_flag,should_use_section_groups",
+        [
+            pytest.param(
+                True,
+                True,
+                True,
+                id="If we're in Canvas and the section_groups feature flag is "
+                "on then should_use_section_groups is True",
+            ),
+            pytest.param(
+                True,
+                False,
+                False,
+                id="If the section_groups feature flag is off then "
+                "should_use_section_groups is False",
+            ),
+            pytest.param(
+                False,
+                True,
+                False,
+                id="If we're not in Canvas then should_use_section_groups is " "False",
+            ),
+            pytest.param(
+                False,
+                True,
+                False,
+                id="If we're not in Canvas and the section_groups feature "
+                "flag is off then should_use_section_groups is False",
+            ),
+        ],
+    )
+    def test_it(
+        self,
+        lti_launch,
+        pyramid_request,
+        is_canvas,
+        feature_flag,
+        should_use_section_groups,
+    ):
+        if feature_flag:
+            pyramid_request.feature.side_effect = (
+                lambda feature: feature == "section_groups"
+            )
+        else:
+            pyramid_request.feature.side_effect = (
+                lambda feature: feature != "section_groups"
+            )
```

Comment:
That doesn't work!

Reply:
Went for this instead:

```python
        def feature(feature_flag):
            if feature_flag == "section_groups":
                return is_feature_flag_enabled
            return mock.DEFAULT

        pyramid_request.feature.side_effect = feature
```

Reply:
What's the boo boo?

Reply:
PyTest ended up pointing out that the `if` above is always taken, so I ended up simplifying this further:

```python
pyramid_request.feature.return_value = is_feature_flag_enabled
```

I think if we want a test mock that _only_ enables the `section_groups` feature flag rather than all feature flags, we'll end up with what I originally had. It's really not necessary though: just setting the boolean return value is fine

Reply:
That's the Gordian knot solution right there. Honestly it's undefined whether we should be returning true for false for the rest anyway.

Reply:
https://github.com/hypothesis/lms/pull/1680/commits/42a52bd474a2079a2593e5ced6195c67874ebaa5